src/tests/encore/forward/forwardInnerForward.enc

```diff
@@ -0,0 +1,32 @@
+active class Base
+  def base() : int
+    42
+  end
+end
+
+active class Bar
+  def bar() : int
+    24
+  end
+end
+
+active class Foo
+  def foo(b : bool) : int
+    val arg1 = (new Bar) ! bar()
+    val arg2 = (new Base) ! base()
+    forward(if b then
+              forward(arg1)
+              arg1
```

Comment:
Since the `forward`'s type is too restrictive, an extra variable `arg1` is necessary. I have added the comment on the unreachable extra argument `arg1`. A new issue will be created correspondingly.